Patch for making ITK 4.5 compatible with samseg.
* use itk::CellGeometryEnum:: instead of AtlasMesh::CellType
* create event class instead of using itkEventMacro
* Set strickter enum scoping rules
* Use ITK v4 registration framework 

Disclaimer: This patch was written with the help of LLMs. The patch itself was not generated by an LLM.
Author: Pavel Tomanek (Inuits), Lara Peeters (UGent)
`` 

diff -ruN samseg-0.5a.orig/gems/cuda/cudatetrahedralmesh.hpp samseg-0.5a/gems/cuda/cudatetrahedralmesh.hpp
--- samseg-0.5a.orig/gems/cuda/cudatetrahedralmesh.hpp	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/cuda/cudatetrahedralmesh.hpp	2026-05-13 11:14:12.178808000 +0200
@@ -119,7 +119,7 @@
 	for( auto cellIt = mesh->GetCells()->Begin();
 	     cellIt != mesh->GetCells()->End();
 	     ++cellIt ) {
-	  if( cellIt.Value()->GetType() == kvl::AtlasMesh::CellType::TETRAHEDRON_CELL ) {
+	  if( cellIt.Value()->GetType() == itk::CellGeometryEnum::TETRAHEDRON_CELL ) {
 	    ids.push_back( cellIt.Index() );
 	  }
 	}
diff -ruN samseg-0.5a.orig/gems/cuda/visitcountersimplecuda.hpp samseg-0.5a/gems/cuda/visitcountersimplecuda.hpp
--- samseg-0.5a.orig/gems/cuda/visitcountersimplecuda.hpp	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/cuda/visitcountersimplecuda.hpp	2026-05-13 11:11:53.176143514 +0200
@@ -44,7 +44,7 @@
 	for( auto cellIt = mesh->GetCells()->Begin();
 	     cellIt != mesh->GetCells()->End();
 	     ++cellIt ) {
-	  if( cellIt.Value()->GetType() == AtlasMesh::CellType::TETRAHEDRON_CELL ) {
+	  if( cellIt.Value()->GetType() == itk::CellGeometryEnum::TETRAHEDRON_CELL ) {
 	    tetrahedronIds.push_back( cellIt.Index() );
 	  }
 	}
diff -ruN samseg-0.5a.orig/gems/Executables/kvlAtlasMeshBuilder.cxx samseg-0.5a/gems/Executables/kvlAtlasMeshBuilder.cxx
--- samseg-0.5a.orig/gems/Executables/kvlAtlasMeshBuilder.cxx	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/Executables/kvlAtlasMeshBuilder.cxx	2026-05-13 11:03:03.902733000 +0200
@@ -473,7 +473,7 @@
         cellIt != miniCollection->GetCells()->End(); ++cellIt )
     {
     const AtlasMesh::CellType*  cell = cellIt.Value();
-    if( cell->GetType() != AtlasMesh::CellType::VERTEX_CELL )
+    if( cell->GetType() != itk::CellGeometryEnum::VERTEX_CELL )
       {
       continue;
       }
@@ -1265,7 +1265,7 @@
   for ( AtlasMesh::CellsContainer::ConstIterator  cellIt = m_Current->GetCells()->Begin();
         cellIt != m_Current->GetCells()->End(); ++cellIt )
     {
-    if ( cellIt.Value()->GetType() == AtlasMesh::CellType::LINE_CELL )
+    if ( cellIt.Value()->GetType() == itk::CellGeometryEnum::LINE_CELL )
       {
       edges.push_back( cellIt.Index() );
       }
diff -ruN samseg-0.5a.orig/gems/Executables/kvlAtlasMeshCollectionFastReferencePositionCost.cxx samseg-0.5a/gems/Executables/kvlAtlasMeshCollectionFastReferencePositionCost.cxx
--- samseg-0.5a.orig/gems/Executables/kvlAtlasMeshCollectionFastReferencePositionCost.cxx	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/Executables/kvlAtlasMeshCollectionFastReferencePositionCost.cxx	2026-05-13 11:17:15.214106000 +0200
@@ -108,7 +108,7 @@
     {
     const AtlasMesh::CellType*  cell = cellIt.Value();
 
-    if( cell->GetType() != AtlasMesh::CellType::TETRAHEDRON_CELL )
+    if( cell->GetType() != itk::CellGeometryEnum::TETRAHEDRON_CELL )
       {
       ++cellIt;
       continue;
diff -ruN samseg-0.5a.orig/gems/Executables/kvlMultiResolutionAtlasMesher.cxx samseg-0.5a/gems/Executables/kvlMultiResolutionAtlasMesher.cxx
--- samseg-0.5a.orig/gems/Executables/kvlMultiResolutionAtlasMesher.cxx	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/Executables/kvlMultiResolutionAtlasMesher.cxx	2026-05-13 11:16:41.599851000 +0200
@@ -401,7 +401,7 @@
         {
         const AtlasMesh::CellType*  cell = cellIt.Value();
 
-        if ( cell->GetType() != AtlasMesh::CellType::TETRAHEDRON_CELL )
+        if ( cell->GetType() != itk::CellGeometryEnum::TETRAHEDRON_CELL )
           {
           continue;
           }
@@ -603,7 +603,7 @@
         {
         const AtlasMesh::CellType*  cell = cellIt.Value();
 
-        if ( cell->GetType() != AtlasMesh::CellType::TETRAHEDRON_CELL )
+        if ( cell->GetType() != itk::CellGeometryEnum::TETRAHEDRON_CELL )
           {
           continue;
           }
diff -ruN samseg-0.5a.orig/gems/GUI/kvlImageViewer.cxx samseg-0.5a/gems/GUI/kvlImageViewer.cxx
--- samseg-0.5a.orig/gems/GUI/kvlImageViewer.cxx	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/GUI/kvlImageViewer.cxx	2026-05-13 11:01:31.384572000 +0200
@@ -744,7 +744,7 @@
         it != mesh->GetCells()->End(); ++it )
     {
     kvl::AtlasMesh::CellType*  cell = it.Value();
-    if( cell->GetType() != kvl::AtlasMesh::CellType::TRIANGLE_CELL )
+    if( cell->GetType() != itk::CellGeometryEnum::TRIANGLE_CELL )
       {
       continue;
       }
@@ -763,7 +763,7 @@
         it != mesh->GetCells()->End(); ++it )
     {
     kvl::AtlasMesh::CellType*  cell = it.Value();
-    if( cell->GetType() != kvl::AtlasMesh::CellType::LINE_CELL )
+    if( cell->GetType() != itk::CellGeometryEnum::LINE_CELL )
       {
       continue;
       }
diff -ruN samseg-0.5a.orig/gems/itkMGHImageIO.cxx samseg-0.5a/gems/itkMGHImageIO.cxx
--- samseg-0.5a.orig/gems/itkMGHImageIO.cxx	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/itkMGHImageIO.cxx	2026-05-13 11:53:29.969534000 +0200
@@ -1,4 +1,3 @@
-
 #include <fstream>
 
 #include "itkMGHImageIO.h"
@@ -142,9 +141,9 @@
     m_Dimensions[2] = uzero;
 
     if ( ByteSwapper<int>::SystemIsBigEndian())
-      m_ByteOrder = BigEndian;
+      m_ByteOrder = CommonEnums::IOByteOrder::BigEndian;
     else
-      m_ByteOrder = LittleEndian;
+      m_ByteOrder = CommonEnums::IOByteOrder::LittleEndian;
 
   }
 
@@ -251,18 +250,18 @@
     switch( type )
       {
       case fs::MRI_UCHAR: 
-	m_ComponentType = UCHAR; break;
+	m_ComponentType = CommonEnums::IOComponent::UCHAR; break;
       case fs::MRI_INT:
-	m_ComponentType = INT; break;
+	m_ComponentType = CommonEnums::IOComponent::INT; break;
       case fs::MRI_FLOAT:
-	m_ComponentType = FLOAT; break;
+	m_ComponentType = CommonEnums::IOComponent::FLOAT; break;
       case fs::MRI_SHORT:
-	m_ComponentType = SHORT; break;
+	m_ComponentType = CommonEnums::IOComponent::SHORT; break;
       case fs::MRI_TENSOR:
-	m_ComponentType = FLOAT; m_NumberOfComponents = 9; break;
+	m_ComponentType = CommonEnums::IOComponent::FLOAT; m_NumberOfComponents = 9; break;
       default:
 	itkExceptionMacro(<<" Unknown data type " << type << " using float by default.");
-	m_ComponentType = FLOAT;
+	m_ComponentType = CommonEnums::IOComponent::FLOAT;
       }
 
     // Next short says whether RAS registration information is good. 
@@ -468,19 +467,19 @@
 
     switch(m_ComponentType)
       {
-      case UCHAR:
+      case CommonEnums::IOComponent::UCHAR:
 	ByteSwapper<unsigned char>::SwapRangeFromSystemToBigEndian((unsigned char*)buffer,
 								   numberOfPixels);
 	break;
-      case SHORT:
+      case CommonEnums::IOComponent::SHORT:
 	ByteSwapper<short>::SwapRangeFromSystemToBigEndian((short*)buffer,
 							   numberOfPixels);
 	break;
-      case INT:
+      case CommonEnums::IOComponent::INT:
 	ByteSwapper<int>::SwapRangeFromSystemToBigEndian((int*)buffer,
 							 numberOfPixels);
 	break;
-      case FLOAT:
+      case CommonEnums::IOComponent::FLOAT:
 	ByteSwapper<float>::SwapRangeFromSystemToBigEndian((float*)buffer,
 							   numberOfPixels);
 	break;
@@ -604,19 +603,19 @@
     unsigned int returnValue = 0;
     switch ( m_ComponentType )
       {
-      case UCHAR: returnValue = sizeof(unsigned char); break;
-      case SHORT: returnValue = sizeof(short); break;
-      case INT:   returnValue = sizeof(int); break;
-      case FLOAT: returnValue = sizeof(float); break;
+      case CommonEnums::IOComponent::UCHAR: returnValue = sizeof(unsigned char); break;
+      case CommonEnums::IOComponent::SHORT: returnValue = sizeof(short); break;
+      case CommonEnums::IOComponent::INT:   returnValue = sizeof(int); break;
+      case CommonEnums::IOComponent::FLOAT: returnValue = sizeof(float); break;
       
       // DJ -- added this in to get the compiler to shut up
-      case UNKNOWNCOMPONENTTYPE:
-      case CHAR:
-      case USHORT:
-      case UINT:
-      case ULONG:
-      case LONG:
-      case DOUBLE:
+      case CommonEnums::IOComponent::UNKNOWNCOMPONENTTYPE:
+      case CommonEnums::IOComponent::CHAR:
+      case CommonEnums::IOComponent::USHORT:
+      case CommonEnums::IOComponent::UINT:
+      case CommonEnums::IOComponent::ULONG:
+      case CommonEnums::IOComponent::LONG:
+      case CommonEnums::IOComponent::DOUBLE:
         break;
       }
     return returnValue;
diff -ruN samseg-0.5a.orig/gems/itkMGHImageIO.h samseg-0.5a/gems/itkMGHImageIO.h
--- samseg-0.5a.orig/gems/itkMGHImageIO.h	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/itkMGHImageIO.h	2026-05-13 10:40:51.201944000 +0200
@@ -1,5 +1,4 @@
 
-
 #ifndef H_ITK_IMAGE_IO_H
 #define H_ITK_IMAGE_IO_H
 
@@ -130,25 +129,25 @@
     // type
     switch( m_ComponentType )
       {
-      case UCHAR:
+      case CommonEnums::IOComponent::UCHAR:
         writer.Write( fs::MRI_UCHAR ); break;
-      case USHORT:
+      case CommonEnums::IOComponent::USHORT:
         itkWarningMacro( << " Casting from USHORT to INT!!!" );
-      case INT:
+      case CommonEnums::IOComponent::INT:
         writer.Write( fs::MRI_INT ); break;
-      case DOUBLE:
+      case CommonEnums::IOComponent::DOUBLE:
         itkWarningMacro( << " Casting from DOUBLE to INT!!!" );
-      case FLOAT:
+      case CommonEnums::IOComponent::FLOAT:
         writer.Write( fs::MRI_FLOAT ); break;
-      case SHORT:
+      case CommonEnums::IOComponent::SHORT:
         writer.Write( fs::MRI_SHORT ); break;
 
       // DJ -- added these cases to make the compiler shut up
-      case UNKNOWNCOMPONENTTYPE:
-      case CHAR:
-      case UINT:
-      case ULONG:
-      case LONG:
+      case CommonEnums::IOComponent::UNKNOWNCOMPONENTTYPE:
+      case CommonEnums::IOComponent::CHAR:
+      case CommonEnums::IOComponent::UINT:
+      case CommonEnums::IOComponent::ULONG:
+      case CommonEnums::IOComponent::LONG:
         break;
 
       }
diff -ruN samseg-0.5a.orig/gems/kvlAtlasMeshCollection.cxx samseg-0.5a/gems/kvlAtlasMeshCollection.cxx
--- samseg-0.5a.orig/gems/kvlAtlasMeshCollection.cxx	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/kvlAtlasMeshCollection.cxx	2026-05-13 11:18:32.907133000 +0200
@@ -197,7 +197,7 @@
       {
       AtlasMesh::CellType*  cell = cellIt.Value();
     
-      if( cell->GetType() == AtlasMesh::CellType::TETRAHEDRON_CELL )
+      if( cell->GetType() == itk::CellGeometryEnum::TETRAHEDRON_CELL )
         {
         // Create info structure and fill in all the fields one by one
         ReferenceTetrahedronInfo  info;
@@ -386,19 +386,19 @@
     
     out << "   " << cellIt.Index() << "   ";
     
-    if ( cell->GetType() == AtlasMesh::CellType::VERTEX_CELL )
+    if ( cell->GetType() == itk::CellGeometryEnum::VERTEX_CELL )
       {
       out << "VERTEX   ";
       }
-    else if ( cell->GetType() == AtlasMesh::CellType::LINE_CELL )
+    else if ( cell->GetType() == itk::CellGeometryEnum::LINE_CELL )
       {
       out << "LINE   ";
       }
-    else if ( cell->GetType() == AtlasMesh::CellType::TRIANGLE_CELL )
+    else if ( cell->GetType() == itk::CellGeometryEnum::TRIANGLE_CELL )
       {
       out << "TRIANGLE   ";
       }
-    else if ( cell->GetType() == AtlasMesh::CellType::TETRAHEDRON_CELL )
+    else if ( cell->GetType() == itk::CellGeometryEnum::TETRAHEDRON_CELL )
       {
       out << "TETRAHEDRON   ";
       }
@@ -1069,7 +1069,7 @@
   if ( !m_Cells->IndexExists( edgeId ) )
     return false;
     
-  if ( m_Cells->ElementAt( edgeId )->GetType() !=  AtlasMesh::CellType::LINE_CELL )
+  if ( m_Cells->ElementAt( edgeId )->GetType() !=  itk::CellGeometryEnum::LINE_CELL )
     return false;
  
   // Retrieve the id's of the two vertices of the edge to be collapsed
@@ -1200,7 +1200,7 @@
     // Get the cell
     const AtlasMesh::CellType*  cell = m_Cells->ElementAt( *disappearingIt );
   
-    if ( cell->GetType() != AtlasMesh::CellType::TETRAHEDRON_CELL )
+    if ( cell->GetType() != itk::CellGeometryEnum::TETRAHEDRON_CELL )
       {
       continue;
       }
@@ -1270,7 +1270,7 @@
         // Get the cell
         const AtlasMesh::CellType*  cell = m_Cells->ElementAt( *it );
       
-        if ( cell->GetType() == AtlasMesh::CellType::LINE_CELL )
+        if ( cell->GetType() == itk::CellGeometryEnum::LINE_CELL )
           {
           //std::cout << "        Found an extra disappearing line that contains both "
           //          << *otherIt << " and p1: " << *it << std::endl;
@@ -1306,7 +1306,7 @@
       // Get the cell
       const AtlasMesh::CellType*  cell = m_Cells->ElementAt( *it );
 
-      if ( cell->GetType() == AtlasMesh::CellType::TRIANGLE_CELL )
+      if ( cell->GetType() == itk::CellGeometryEnum::TRIANGLE_CELL )
         {
         //std::cout << "        Found an extra disappearing triangle that contains "
         //          << firstOtherPointId << ", " << secondOtherPointId << ", and p1: " << *it << std::endl;
@@ -1327,7 +1327,7 @@
     // Get the cell
     const AtlasMesh::CellType*  cell = m_Cells->ElementAt( *it );
 
-    if ( cell->GetType() == AtlasMesh::CellType::VERTEX_CELL )
+    if ( cell->GetType() == itk::CellGeometryEnum::VERTEX_CELL )
       {
       //std::cout << "        Found an extra disappearing vertex that contains p1" << std::endl;
       extraDisappearingCells.insert( *it );
@@ -1359,7 +1359,7 @@
     // Get the cell
     const AtlasMesh::CellType*  cell = m_Cells->ElementAt( *p0orP1It );
   
-    if ( cell->GetType() != AtlasMesh::CellType::TETRAHEDRON_CELL )
+    if ( cell->GetType() != itk::CellGeometryEnum::TETRAHEDRON_CELL )
       {
       continue;
       }
@@ -1576,7 +1576,7 @@
     // Get the cell
     const AtlasMesh::CellType*  cell = m_Cells->ElementAt( *it );
 
-    if ( cell->GetType() == AtlasMesh::CellType::VERTEX_CELL )
+    if ( cell->GetType() == itk::CellGeometryEnum::VERTEX_CELL )
       {
       unifiedVertexId = *it;
       //std::cout << "        Found that unifiedVertexId is " << unifiedVertexId << std::endl;
@@ -1778,17 +1778,17 @@
 
       AtlasMesh::CellAutoPointer newCell;
       
-      if ( cell->GetType() == AtlasMesh::CellType::VERTEX_CELL )
+      if ( cell->GetType() == itk::CellGeometryEnum::VERTEX_CELL )
         {
         // Create a new vertex cell
         newCell.TakeOwnership( new VertexCell );
         }
-      else if ( cell->GetType() == AtlasMesh::CellType::LINE_CELL  )
+      else if ( cell->GetType() == itk::CellGeometryEnum::LINE_CELL  )
         {
         // Create a new line cell
         newCell.TakeOwnership( new LineCell );
         }
-      else if ( cell->GetType() == AtlasMesh::CellType::TRIANGLE_CELL  )
+      else if ( cell->GetType() == itk::CellGeometryEnum::TRIANGLE_CELL  )
         {
         // Create a new triangle cell
         newCell.TakeOwnership( new TriangleCell );
@@ -1941,17 +1941,17 @@
       typedef itk::TetrahedronCell< AtlasMesh::CellType >  TetrahedronCell;
 
       AtlasMesh::CellAutoPointer newCell;
-      if ( cell->GetType() == AtlasMesh::CellType::VERTEX_CELL )
+      if ( cell->GetType() == itk::CellGeometryEnum::VERTEX_CELL )
         {
         // Create a new vertex cell
         newCell.TakeOwnership( new VertexCell );
         }
-      else if ( cell->GetType() == AtlasMesh::CellType::LINE_CELL )
+      else if ( cell->GetType() == itk::CellGeometryEnum::LINE_CELL )
         {
         // Create a new line cell
         newCell.TakeOwnership( new LineCell );
         }
-      else if ( cell->GetType() == AtlasMesh::CellType::TRIANGLE_CELL )
+      else if ( cell->GetType() == itk::CellGeometryEnum::TRIANGLE_CELL )
         {
         // Create a new triangle cell
         newCell.TakeOwnership( new TriangleCell );
@@ -2020,7 +2020,7 @@
         typedef itk::TetrahedronCell< AtlasMesh::CellType >  TetrahedronCell;
 
         AtlasMesh::CellAutoPointer newCell;
-        if ( cell->GetType() == AtlasMesh::CellType::LINE_CELL  )
+        if ( cell->GetType() == itk::CellGeometryEnum::LINE_CELL  )
           {
           // Create a new line cell
           newCell.TakeOwnership( new LineCell );
@@ -2045,12 +2045,12 @@
           //std::cout << " to the outer points." << std::endl;
           
           }
-        else if ( cell->GetType() == AtlasMesh::CellType::TRIANGLE_CELL )
+        else if ( cell->GetType() == itk::CellGeometryEnum::TRIANGLE_CELL )
           {
           // Create a new triangle cell
           newCell.TakeOwnership( new TriangleCell );
           }
-        else if ( cell->GetType() == AtlasMesh::CellType::TETRAHEDRON_CELL )
+        else if ( cell->GetType() == itk::CellGeometryEnum::TETRAHEDRON_CELL )
           {
           // Create a new tetrahedron cell
           newCell.TakeOwnership( new TetrahedronCell );
@@ -2086,7 +2086,7 @@
         {
         const AtlasMesh::CellType*  cell = m_Cells->ElementAt( *neighborIt );
 
-        if ( cell->GetType() == AtlasMesh::CellType::VERTEX_CELL )
+        if ( cell->GetType() == itk::CellGeometryEnum::VERTEX_CELL )
           {
           // Create a new cell of the correct type
           typedef itk::VertexCell< AtlasMesh::CellType >    VertexCell;
@@ -2101,7 +2101,7 @@
           regionGrownCells->InsertElement( *neighborIt, newCell.ReleaseOwnership() );
           
           }
-        else if ( cell->GetType() == AtlasMesh::CellType::LINE_CELL )
+        else if ( cell->GetType() == itk::CellGeometryEnum::LINE_CELL )
           {
           // Retrieve the other point of this line
           AtlasMesh::CellType::PointIdConstIterator  pointIt = cell->PointIdsBegin();
@@ -2143,7 +2143,7 @@
             
                       
           }
-        else if ( cell->GetType() == AtlasMesh::CellType::TRIANGLE_CELL )
+        else if ( cell->GetType() == itk::CellGeometryEnum::TRIANGLE_CELL )
           {
           // Retrieve the other two points of this triangle
           AtlasMesh::CellType::PointIdConstIterator  pointIt = cell->PointIdsBegin();
@@ -2253,7 +2253,7 @@
     {
     AtlasMesh::CellType*  cell = cellIt.Value();
     
-    if ( cell->GetType() == AtlasMesh::CellType::VERTEX_CELL )
+    if ( cell->GetType() == itk::CellGeometryEnum::VERTEX_CELL )
       {
       // Retrieve point id
       AtlasMesh::PointIdentifier  pointId = *( cell->PointIdsBegin() );
@@ -2378,7 +2378,7 @@
       {
       const AtlasMesh::CellType*  cell = cellIt.Value();
   
-      if( cell->GetType() != AtlasMesh::CellType::TETRAHEDRON_CELL )
+      if( cell->GetType() != itk::CellGeometryEnum::TETRAHEDRON_CELL )
         {
         ++cellIt;
         continue;
@@ -2488,7 +2488,7 @@
       {
       const AtlasMesh::CellType*  cell = cellIt.Value();
   
-      if ( cell->GetType() != AtlasMesh::CellType::TETRAHEDRON_CELL )
+      if ( cell->GetType() != itk::CellGeometryEnum::TETRAHEDRON_CELL )
         {
         continue;
         }
@@ -3135,7 +3135,7 @@
   if ( !m_Cells->IndexExists( edgeId ) )
     return 0;
     
-  if ( m_Cells->ElementAt( edgeId )->GetType() !=  AtlasMesh::CellType::LINE_CELL )
+  if ( m_Cells->ElementAt( edgeId )->GetType() !=  itk::CellGeometryEnum::LINE_CELL )
     return 0;
 
   // Edges on borders are typically degenerate when swapped; simply forbid such swaps
@@ -3274,7 +3274,7 @@
   if ( !m_Cells->IndexExists( edgeId ) )
     return 0;
     
-  if ( m_Cells->ElementAt( edgeId )->GetType() !=  AtlasMesh::CellType::LINE_CELL )
+  if ( m_Cells->ElementAt( edgeId )->GetType() !=  itk::CellGeometryEnum::LINE_CELL )
     return 0;
 
 
@@ -3388,7 +3388,7 @@
     {
     const AtlasMesh::CellType*  cell = cellIt.Value();
     
-    if ( cell->GetType() == AtlasMesh::CellType::VERTEX_CELL )
+    if ( cell->GetType() == itk::CellGeometryEnum::VERTEX_CELL )
       {
       // Copy vertex cell
       AtlasMesh::CellAutoPointer  newVertex;
@@ -3397,7 +3397,7 @@
       newVertex->SetPointId( 0, *pointIt );
       splittedCells->InsertElement( cellIt.Index(), newVertex.ReleaseOwnership() );
       }
-    else if ( cell->GetType() == AtlasMesh::CellType::LINE_CELL  )
+    else if ( cell->GetType() == itk::CellGeometryEnum::LINE_CELL  )
       {
       // Retrieve point ids of the vertices
       AtlasMesh::CellType::PointIdConstIterator  pointIt = cell->PointIdsBegin();
diff -ruN samseg-0.5a.orig/gems/kvlAtlasMeshCollectionValidator.cxx samseg-0.5a/gems/kvlAtlasMeshCollectionValidator.cxx
--- samseg-0.5a.orig/gems/kvlAtlasMeshCollectionValidator.cxx	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/kvlAtlasMeshCollectionValidator.cxx	2026-05-13 11:15:13.250064000 +0200
@@ -73,7 +73,7 @@
   for ( AtlasMeshCollection::CellsContainerType::ConstIterator cellIt = meshCollection->GetCells()->Begin();
         cellIt != meshCollection->GetCells()->End(); ++cellIt )
     {
-    if ( cellIt.Value()->GetType() != AtlasMesh::CellType::TETRAHEDRON_CELL )
+    if ( cellIt.Value()->GetType() != itk::CellGeometryEnum::TETRAHEDRON_CELL )
       {
       continue;
       }
@@ -110,15 +110,15 @@
 
       if ( allThePointsAreGood )
         {
-        if ( cellIt2.Value()->GetType() == AtlasMesh::CellType::VERTEX_CELL )
+        if ( cellIt2.Value()->GetType() == itk::CellGeometryEnum::VERTEX_CELL )
           {
           numberOfVerticesFound++;
           }
-        else if ( cellIt2.Value()->GetType() == AtlasMesh::CellType::LINE_CELL )
+        else if ( cellIt2.Value()->GetType() == itk::CellGeometryEnum::LINE_CELL )
           {
           numberOfLinesFound++;
           }
-        else if ( cellIt2.Value()->GetType() == AtlasMesh::CellType::TRIANGLE_CELL )
+        else if ( cellIt2.Value()->GetType() == itk::CellGeometryEnum::TRIANGLE_CELL )
           {
           numberOfTrianglesFound++;
           }
@@ -152,7 +152,7 @@
   for ( AtlasMeshCollection::CellsContainerType::ConstIterator cellIt = meshCollection->GetCells()->Begin();
         cellIt != meshCollection->GetCells()->End(); ++cellIt )
     {
-    if ( cellIt.Value()->GetType() != AtlasMesh::CellType::TRIANGLE_CELL )
+    if ( cellIt.Value()->GetType() != itk::CellGeometryEnum::TRIANGLE_CELL )
       {
       continue;
       }
@@ -186,11 +186,11 @@
 
       if ( allThePointsAreGood )
         {
-        if ( cellIt2.Value()->GetType() == AtlasMesh::CellType::VERTEX_CELL )
+        if ( cellIt2.Value()->GetType() == itk::CellGeometryEnum::VERTEX_CELL )
           {
           numberOfVerticesFound++;
           }
-        else if ( cellIt2.Value()->GetType() == AtlasMesh::CellType::LINE_CELL )
+        else if ( cellIt2.Value()->GetType() == itk::CellGeometryEnum::LINE_CELL )
           {
           numberOfLinesFound++;
           }
@@ -217,7 +217,7 @@
   for ( AtlasMeshCollection::CellsContainerType::ConstIterator cellIt = meshCollection->GetCells()->Begin();
         cellIt != meshCollection->GetCells()->End(); ++cellIt )
     {
-    if ( cellIt.Value()->GetType() != AtlasMesh::CellType::LINE_CELL )
+    if ( cellIt.Value()->GetType() != itk::CellGeometryEnum::LINE_CELL )
       {
       continue;
       }
@@ -248,7 +248,7 @@
 
       if ( allThePointsAreGood )
         {
-        if ( cellIt2.Value()->GetType() == AtlasMesh::CellType::VERTEX_CELL )
+        if ( cellIt2.Value()->GetType() == itk::CellGeometryEnum::VERTEX_CELL )
           {
           numberOfVerticesFound++;
           }
@@ -300,7 +300,7 @@
   for ( AtlasMeshCollection::CellsContainerType::ConstIterator cellIt = meshCollection->GetCells()->Begin();
         cellIt != meshCollection->GetCells()->End(); ++cellIt )
     {
-    if ( cellIt.Value()->GetType() != AtlasMesh::CellType::TETRAHEDRON_CELL )
+    if ( cellIt.Value()->GetType() != itk::CellGeometryEnum::TETRAHEDRON_CELL )
       {
       continue;
       }
diff -ruN samseg-0.5a.orig/gems/kvlAtlasMeshDeformationOptimizer.h samseg-0.5a/gems/kvlAtlasMeshDeformationOptimizer.h
--- samseg-0.5a.orig/gems/kvlAtlasMeshDeformationOptimizer.h	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/kvlAtlasMeshDeformationOptimizer.h	2026-05-11 16:45:11.252024924 +0200
@@ -1,6 +1,7 @@
 #ifndef __kvlAtlasMeshDeformationOptimizer_h
 #define __kvlAtlasMeshDeformationOptimizer_h
 
+#include "itkEventObject.h"
 #include "kvlAtlasMesh.h"
 #include "kvlAtlasMeshPositionCostAndGradientCalculator.h"
 
@@ -9,9 +10,47 @@
 {
 
 // Events generated
-itkEventMacro( DeformationStartEvent, itk::UserEvent );
-itkEventMacro( DeformationIterationEvent, itk::UserEvent );
-itkEventMacro( DeformationEndEvent, itk::UserEvent );
+class DeformationStartEvent : public itk::UserEvent
+{
+public:
+  typedef DeformationStartEvent Self;
+  DeformationStartEvent() {}
+  virtual ~DeformationStartEvent() {}
+  const char* GetEventName() const override
+    { return "DeformationStartEvent"; }
+  bool CheckEvent(const ::itk::EventObject* e) const override
+    { return dynamic_cast<const Self*>(e); }
+  ::itk::EventObject* MakeObject() const override
+    { return new Self(*this); }
+};
+
+class DeformationIterationEvent : public itk::UserEvent
+{
+public:
+  typedef DeformationIterationEvent Self;
+  DeformationIterationEvent() {}
+  virtual ~DeformationIterationEvent() {}
+  const char* GetEventName() const override
+    { return "DeformationIterationEvent"; }
+  bool CheckEvent(const ::itk::EventObject* e) const override
+    { return dynamic_cast<const Self*>(e); }
+  ::itk::EventObject* MakeObject() const override
+    { return new Self(*this); }
+};
+
+class DeformationEndEvent : public itk::UserEvent
+{
+public:
+  typedef DeformationEndEvent Self;
+  DeformationEndEvent() {}
+  virtual ~DeformationEndEvent() {}
+  const char* GetEventName() const override
+    { return "DeformationEndEvent"; }
+  bool CheckEvent(const ::itk::EventObject* e) const override
+    { return dynamic_cast<const Self*>(e); }
+  ::itk::EventObject* MakeObject() const override
+    { return new Self(*this); }
+};
 
 
 /**
diff -ruN samseg-0.5a.orig/gems/kvlAtlasMeshRasterizor.cxx samseg-0.5a/gems/kvlAtlasMeshRasterizor.cxx
--- samseg-0.5a.orig/gems/kvlAtlasMeshRasterizor.cxx	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/kvlAtlasMeshRasterizor.cxx	2026-05-13 11:12:59.640668000 +0200
@@ -44,7 +44,7 @@
   for ( AtlasMesh::CellsContainer::ConstIterator  cellIt = mesh->GetCells()->Begin();
         cellIt != mesh->GetCells()->End(); ++cellIt )
     {
-    if ( cellIt.Value()->GetType() == AtlasMesh::CellType::TETRAHEDRON_CELL )
+    if ( cellIt.Value()->GetType() == itk::CellGeometryEnum::TETRAHEDRON_CELL )
       {
       str.m_TetrahedronIds.push_back( cellIt.Index() );
       //str.m_TetrahedronIds.insert( cellIt.Index() );
diff -ruN samseg-0.5a.orig/gems/kvlAtlasParameterEstimator.h samseg-0.5a/gems/kvlAtlasParameterEstimator.h
--- samseg-0.5a.orig/gems/kvlAtlasParameterEstimator.h	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/kvlAtlasParameterEstimator.h	2026-05-11 16:42:49.469797515 +0200
@@ -1,6 +1,7 @@
 #ifndef __kvlAtlasParameterEstimator_h
 #define __kvlAtlasParameterEstimator_h
 
+#include "itkEventObject.h"
 #include "itkImage.h"
 #include "kvlAtlasMeshCollection.h"
 #include "kvlCompressionLookupTable.h"
@@ -10,12 +11,89 @@
 {
 
 // Events generated
-itkEventMacro( AlphasEstimationStartEvent, itk::UserEvent );
-itkEventMacro( AlphasEstimationIterationEvent, itk::UserEvent );
-itkEventMacro( AlphasEstimationEndEvent, itk::UserEvent );
-itkEventMacro( PositionEstimationStartEvent, itk::UserEvent );
-itkEventMacro( PositionEstimationIterationEvent, itk::UserEvent );
-itkEventMacro( PositionEstimationEndEvent, itk::UserEvent );
+class AlphasEstimationStartEvent : public itk::UserEvent
+{
+public:
+  typedef AlphasEstimationStartEvent Self;
+  AlphasEstimationStartEvent() {}
+  virtual ~AlphasEstimationStartEvent() {}
+  const char* GetEventName() const override
+    { return "AlphasEstimationStartEvent"; }
+  bool CheckEvent(const ::itk::EventObject* e) const override
+    { return dynamic_cast<const Self*>(e); }
+  ::itk::EventObject* MakeObject() const override
+    { return new Self(*this); }
+};
+
+class AlphasEstimationIterationEvent : public itk::UserEvent
+{
+public:
+  typedef AlphasEstimationIterationEvent Self;
+  AlphasEstimationIterationEvent() {}
+  virtual ~AlphasEstimationIterationEvent() {}
+  const char* GetEventName() const override
+    { return "AlphasEstimationIterationEvent"; }
+  bool CheckEvent(const ::itk::EventObject* e) const override
+    { return dynamic_cast<const Self*>(e); }
+  ::itk::EventObject* MakeObject() const override
+    { return new Self(*this); }
+};
+
+class AlphasEstimationEndEvent : public itk::UserEvent
+{
+public:
+  typedef AlphasEstimationEndEvent Self;
+  AlphasEstimationEndEvent() {}
+  virtual ~AlphasEstimationEndEvent() {}
+  const char* GetEventName() const override
+    { return "AlphasEstimationEndEvent"; }
+  bool CheckEvent(const ::itk::EventObject* e) const override
+    { return dynamic_cast<const Self*>(e); }
+  ::itk::EventObject* MakeObject() const override
+    { return new Self(*this); }
+};
+
+class PositionEstimationStartEvent : public itk::UserEvent
+{
+public:
+  typedef PositionEstimationStartEvent Self;
+  PositionEstimationStartEvent() {}
+  virtual ~PositionEstimationStartEvent() {}
+  const char* GetEventName() const override
+    { return "PositionEstimationStartEvent"; }
+  bool CheckEvent(const ::itk::EventObject* e) const override
+    { return dynamic_cast<const Self*>(e); }
+  ::itk::EventObject* MakeObject() const override
+    { return new Self(*this); }
+};
+
+class PositionEstimationIterationEvent : public itk::UserEvent
+{
+public:
+  typedef PositionEstimationIterationEvent Self;
+  PositionEstimationIterationEvent() {}
+  virtual ~PositionEstimationIterationEvent() {}
+  const char* GetEventName() const override
+    { return "PositionEstimationIterationEvent"; }
+  bool CheckEvent(const ::itk::EventObject* e) const override
+    { return dynamic_cast<const Self*>(e); }
+  ::itk::EventObject* MakeObject() const override
+    { return new Self(*this); }
+};
+
+class PositionEstimationEndEvent : public itk::UserEvent
+{
+public:
+  typedef PositionEstimationEndEvent Self;
+  PositionEstimationEndEvent() {}
+  virtual ~PositionEstimationEndEvent() {}
+  const char* GetEventName() const override
+    { return "PositionEstimationEndEvent"; }
+  bool CheckEvent(const ::itk::EventObject* e) const override
+    { return dynamic_cast<const Self*>(e); }
+  ::itk::EventObject* MakeObject() const override
+    { return new Self(*this); }
+};
 
 
 class AtlasParameterEstimator: public itk::Object
diff -ruN samseg-0.5a.orig/gems/kvlCroppedImageReader.cxx samseg-0.5a/gems/kvlCroppedImageReader.cxx
--- samseg-0.5a.orig/gems/kvlCroppedImageReader.cxx	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/kvlCroppedImageReader.cxx	2026-05-11 16:47:29.012907995 +0200
@@ -133,7 +133,7 @@
 
   // Get someone who can read this file format
   itk::ImageIOBase::Pointer  io = itk::ImageIOFactory::CreateImageIO(  filename.c_str(), 
-                                                                       itk::ImageIOFactory::ReadMode );
+                                                                       itk::ImageIOFactory::IOFileModeEnum::ReadMode );
   if ( io )
     {
     std::cout << "Constructing image-to-world transform from header information ("
@@ -268,7 +268,7 @@
 
     // Obtain the bounding box size
     itk::ImageIOBase::Pointer   imageIO =
-              itk::ImageIOFactory::CreateImageIO( boundingFileName, itk::ImageIOFactory::ReadMode );
+              itk::ImageIOFactory::CreateImageIO( boundingFileName, itk::ImageIOFactory::IOFileModeEnum::ReadMode );
     if ( imageIO.IsNull() )
       {
       itkExceptionMacro( << "Can't find a suitable filter to import image " << boundingFileName );
diff -ruN samseg-0.5a.orig/gems/kvlRegisterImages.hxx samseg-0.5a/gems/kvlRegisterImages.hxx
--- samseg-0.5a.orig/gems/kvlRegisterImages.hxx	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/kvlRegisterImages.hxx	2026-05-11 16:49:32.187535383 +0200
@@ -175,7 +175,7 @@
     registration->MetricSamplingReinitializeSeed(randomNumberGeneratorSeed);
 
     //Set sampling strategy
-    typename RegistrationType::MetricSamplingStrategyType samplingStrategy = RegistrationType::RANDOM;
+    typename RegistrationType::MetricSamplingStrategyEnum samplingStrategy = RegistrationType::MetricSamplingStrategyEnum::RANDOM;
     registration->SetMetricSamplingStrategy(samplingStrategy);
     registration->SetMetricSamplingPercentage(m_SamplingPercentage);
 
diff -ruN samseg-0.5a.orig/gems/Matlab/kvlChangeK.h samseg-0.5a/gems/Matlab/kvlChangeK.h
--- samseg-0.5a.orig/gems/Matlab/kvlChangeK.h	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/Matlab/kvlChangeK.h	2026-05-13 11:08:33.362312086 +0200
@@ -75,7 +75,7 @@
       {
 
 	AtlasMesh::CellType*  cell = cellIt.Value();
-	if( cell->GetType() == AtlasMesh::CellType::TETRAHEDRON_CELL ){
+	if( cell->GetType() == itk::CellGeometryEnum::TETRAHEDRON_CELL ){
 	  
 	  AtlasMesh::CellType::PointIdIterator  pit = cell->PointIdsBegin();
 	  AtlasAlphasType alphas0  = mesh->GetPointData()->ElementAt(*pit).m_Alphas;
diff -ruN samseg-0.5a.orig/gems/Matlab/kvlReadMeshCollection.h samseg-0.5a/gems/Matlab/kvlReadMeshCollection.h
--- samseg-0.5a.orig/gems/Matlab/kvlReadMeshCollection.h	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/Matlab/kvlReadMeshCollection.h	2026-05-13 11:19:54.000567000 +0200
@@ -116,7 +116,7 @@
           {
           kvl::AtlasMesh::CellType*  cell = cellIt.Value();
 
-          if( cell->GetType() != kvl::AtlasMesh::CellType::TETRAHEDRON_CELL )
+          if( cell->GetType() != itk::CellGeometryEnum::TETRAHEDRON_CELL )
             {
             continue;
             }
diff -ruN samseg-0.5a.orig/gems/Matlab/kvlReadMesh.h samseg-0.5a/gems/Matlab/kvlReadMesh.h
--- samseg-0.5a.orig/gems/Matlab/kvlReadMesh.h	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/Matlab/kvlReadMesh.h	2026-05-13 11:10:08.479537000 +0200
@@ -124,7 +124,7 @@
           {
           kvl::AtlasMesh::CellType*  cell = cellIt.Value();
 
-          if( cell->GetType() != kvl::AtlasMesh::CellType::TETRAHEDRON_CELL )
+          if( cell->GetType() != itk::CellGeometryEnum::TETRAHEDRON_CELL )
             {
             continue;
             }
diff -ruN samseg-0.5a.orig/gems/Matlab/kvlTransformMeshCollection.h samseg-0.5a/gems/Matlab/kvlTransformMeshCollection.h
--- samseg-0.5a.orig/gems/Matlab/kvlTransformMeshCollection.h	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/Matlab/kvlTransformMeshCollection.h	2026-05-13 11:19:19.314879000 +0200
@@ -87,7 +87,7 @@
         {
         kvl::AtlasMesh::CellType*  cell = cellIt.Value();
 
-        if( cell->GetType() != kvl::AtlasMesh::CellType::TETRAHEDRON_CELL )
+        if( cell->GetType() != itk::CellGeometryEnum::TETRAHEDRON_CELL )
           {
           continue;
           }
diff -ruN samseg-0.5a.orig/gems/Testing/testatlasmeshalphadrawer.cpp samseg-0.5a/gems/Testing/testatlasmeshalphadrawer.cpp
--- samseg-0.5a.orig/gems/Testing/testatlasmeshalphadrawer.cpp	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/gems/Testing/testatlasmeshalphadrawer.cpp	2026-05-13 11:15:58.108380000 +0200
@@ -209,7 +209,7 @@
   for( auto cellIt = mesh->GetCells()->Begin();
        cellIt != mesh->GetCells()->End();
        ++cellIt ) {
-    if( cellIt.Value()->GetType() == kvl::AtlasMesh::CellType::TETRAHEDRON_CELL ) {
+    if( cellIt.Value()->GetType() == itk::CellGeometryEnum::TETRAHEDRON_CELL ) {
       tetrahedronIds.push_back( cellIt.Index() );
     } else {
       nOther++;
diff -ruN samseg-0.5a.orig/samseg/cxx/pyKvlMesh.cxx samseg-0.5a/samseg/cxx/pyKvlMesh.cxx
--- samseg-0.5a.orig/samseg/cxx/pyKvlMesh.cxx	2025-05-07 18:18:31.000000000 +0200
+++ samseg-0.5a/samseg/cxx/pyKvlMesh.cxx	2026-05-12 12:41:44.525066000 +0200
@@ -13,6 +13,7 @@
 #include "kvlAtlasMeshProbabilityImageStatisticsCollector.h"
 #include "kvlAtlasMeshJacobianDeterminantDrawer.h"
 #include "itkImageRegionIterator.h"
+#include "itkTetrahedronCell.h"
 
 
 #define XYZ_DIMENSIONS 3
@@ -175,7 +176,7 @@
     {
         kvl::AtlasMesh::CellType*  cell = cellIt.Value();
 
-        if( cell->GetType() != kvl::AtlasMesh::CellType::TETRAHEDRON_CELL )
+        if( cell->GetType() != itk::CellGeometryEnum::TETRAHEDRON_CELL )
         {
             continue;
         }
@@ -816,7 +817,7 @@
       //  
       const kvl::AtlasMesh::CellIdentifier  cellId = *cellIt;
       const kvl::AtlasMesh::CellType*  cell = mesh->GetCells()->ElementAt( cellId );
-      if ( cell->GetType() != kvl::AtlasMesh::CellType::TETRAHEDRON_CELL )
+      if ( cell->GetType() != itk::CellGeometryEnum::TETRAHEDRON_CELL )
         {
         //  
         continue;  
